presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java

```diff
@@ -34,10 +35,17 @@
 public class SpatialJoinUtils
 {
     public static final FullyQualifiedName ST_CONTAINS = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_contains");
-    public static final FullyQualifiedName ST_WITHIN = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_within");
+    public static final FullyQualifiedName ST_CROSSES = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_crosses");
+    public static final FullyQualifiedName ST_EQUALS = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_equals");
     public static final FullyQualifiedName ST_INTERSECTS = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_intersects");
+    public static final FullyQualifiedName ST_OVERLAPS = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_overlaps");
+    public static final FullyQualifiedName ST_TOUCHES = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_touches");
+    public static final FullyQualifiedName ST_WITHIN = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_within");
     public static final FullyQualifiedName ST_DISTANCE = FullyQualifiedName.of(DEFAULT_NAMESPACE, "st_distance");
 
+    private static final List<FullyQualifiedName> ALLOWED_SPATIAL_JOIN_FUNCTIONS = ImmutableList.of(
```

Comment:
this is used only to check if function name is valid; consider changing to a Set<String> to speed up these checks below